azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource_test.go

```diff
@@ -248,6 +248,25 @@ func TestAccEventGridEventSubscription_advancedFilterMaxItems(t *testing.T) {
 	})
 }
 
+func TestAccEventGridEventSubscription_identity(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_eventgrid_event_subscription", "test")
+	r := EventGridEventSubscriptionResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.identity(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("delivery_identity.#").HasValue("1"),
+				check.That(data.ResourceName).Key("delivery_identity.0.type").HasValue("SystemAssigned"),
+				check.That(data.ResourceName).Key("dead_letter_identity.#").HasValue("1"),
+				check.That(data.ResourceName).Key("dead_letter_identity.0.type").HasValue("SystemAssigned"),
+			),
+		},
+		data.ImportStep(),
```

Comment:
since this isn't ForceNew it can be updated - so could we add an update to remove it and then an update to add it:

```suggestion
		data.ImportStep(),
		{
 			Config: r.basic(data),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("delivery_identity.#").HasValue("0"),
 				check.That(data.ResourceName).Key("dead_letter_identity.#").HasValue("0"),
 			),
 		},
 		data.ImportStep(),
		{
 			Config: r.identity(data),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("delivery_identity.#").HasValue("1"),
 				check.That(data.ResourceName).Key("delivery_identity.0.type").HasValue("SystemAssigned"),
 				check.That(data.ResourceName).Key("dead_letter_identity.#").HasValue("1"),
 				check.That(data.ResourceName).Key("dead_letter_identity.0.type").HasValue("SystemAssigned"),
 			),
 		},
 		data.ImportStep(),
```